### PR TITLE
Add file watcher for development with live reload

### PR DESCRIPTION
**Pull request by goncalotomas**
- [x] Make `mix grf.server` build when starting
- [x] Watch for changes in the input directory and call `mix grf.build` every time a file is modified.
- [x] Watch for changes in the layouts directory and call `mix grf.build` every time a file is modified.
- [x] Use `plug_live_reload` to reload pages automatically when changes are detected in the output files.
- [x] Rewrite HTTP Server to use `Plug.Router` but not `Plug.Static` so that Live Reload works
- [x] Fix or suppress lint issues (we are now using credo, had forgotten to add it)
- [x] Remove all Application related code, spawn required processes from within `grf.server`
- [x] Allow changing HTTP port via `-p`, `--port` CLI option

This makes it so that when a user changes a file, it will first be picked up by the file watcher, rebuilt into a new version of the file in the output directory which is then picked up by `plug_live_reload`.

There was an issue with the old way `grf.server` was being started that was not allowing live reload to happen which I still need to dig into a bit more.

### CODE REVIEW

File: lib/web/plug.ex [hunk 1]
```diff
@@ -1,33 +1,43 @@
 defmodule GriffinSSG.Web.Plug do
```
Comment (goncalotomas) [inline lib/web/plug.ex:1]: rewrite

**Comment by goncalotomas** [comment]
Things are starting to take shape but there are still a couple of things to do:

- [x] Make `Watcher` watch for changes in layouts directory aside from input directory
- [x] Rewrite HTTP Server to use `Plug.Router` but not `Plug.Static` so that Live Reload works
- [x] Fix or suppress lint issues (we are now using credo, had forgotten to add it)

**Comment by goncalotomas** [comment]
@lcmen perhaps not too late, but we finally have a server that watches file changes and it even live reloads! 😄

**Comment by lcmen** [comment]
@goncalotomas thanks for the heads-up! Unfortunatelly, I've settled on Jekyll for now.

**Pull request merged**